backend/app/utils/data_fetchers.py

```
@@ -168,7 +169,69 @@ def load_data(self, symbol: str, source: str = "binance", days: int = 365) -> pd
 
 
-def get_data_fetcher(source: str = "coingecko"):
+class CoinMarketCapFetcher(CryptoDataFetcher):
+    """Fetcher for CoinMarketCap API"""
+
+    def __init__(self, api_key: Optional[str] = None):
+        super().__init__()
+        self.api_key = api_key
+        self.base_url = "https://pro-api.coinmarketcap.com/v1"
+
+    def load_data(
```

Pylint (3.3.7)

[refactor] Too many local variables (17/15)

(R0914)